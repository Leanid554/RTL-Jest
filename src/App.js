import { useEffect, useState } from "react";
import "./App.css";
import Clicker from "./components/Clicker";
import LabelBox from "./components/LabelBox";

function App() {
  const [data, setData] = useState(null);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState("");

  const onClick = () => setToggle((prev) => !prev);
  useEffect(() => {
    setTimeout(() => {
      setData({});
    }, 500);
  }, []);
  return (
    <div className="App">
      <LabelBox />
      <Clicker />
      <div>
        <h1 data-testid="value-elem">{value}</h1>
        {toggle === true && <div data-testid="toggle-elem">toggle</div>}
        {data && <div style={{ color: "red" }}>data</div>}
        <h1>Hello world</h1>
        <button data-testid="toggle-btn" onClick={onClick}>
          click me
        </button>
        <input
          onChange={(e) => setValue(e.target.value)}
          type="text"
          placeholder="input value......"
        ></input>
      </div>
    </div>
  );
}

export default App;
