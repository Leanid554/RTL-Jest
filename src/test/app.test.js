import { screen, render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "../App";

describe("TEST APP", () => {
  test("check button and text", () => {
    render(<App />);
    const btn = screen.getByRole("button", { name: /click me/i });
    const text = screen.queryByText(/hello world/i);
    const input = screen.getByPlaceholderText(/input value/i);
    expect(btn).toBeInTheDocument();
    expect(text).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test("check button", async () => {
    render(<App />);
    const HelloWorld = await screen.findByText(/data/i);
    expect(HelloWorld).toBeInTheDocument();
    expect(HelloWorld).toHaveStyle({ color: "red" });
    screen.debug();
  });

  test("Click event", () => {
    render(<App />);
    const input = screen.getByPlaceholderText(/input value/i);
    expect(screen.queryByTestId("value-elem")).toContainHTML("");
    //fireEvent.change(input, {
    //  target: { value: "123123" },
    //});
    userEvent.type(input, "123123");
    expect(screen.queryByTestId("value-elem")).toContainHTML("123123");
  });
});
