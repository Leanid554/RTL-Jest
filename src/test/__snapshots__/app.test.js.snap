// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP check button and text 1`] = `
<input
  placeholder="input value......"
  type="text"
/>
`;
